src/queries/label-text.js

```diff
@@ -12,30 +12,73 @@ import {
 } from './all-utils'
 import {queryAllByText} from './text'
 
+function getCombinations(labels, matcher) {
+  const combs = [[]]
+  const matching = []
+  for (const label of labels) {
+    const copy = [...combs] // See note below.
+    for (const prefix of copy) {
+      const combination = prefix.concat(label.textToMatch)
+      combs.push(combination)
+      if (matcher(combination.join(' '), label.node)) {
+        matching.push(label.node)
+      }
+    }
+  }
+  return matching
+}
+function queryAllLabels(container) {
+  return Array.from(container.querySelectorAll('label,input'))
+    .map(node => {
+      let textToMatch =
+        node.tagName.toLowerCase() === 'label'
+          ? node.textContent
+          : node.value || null
+      // The children of a textarea are part of `textContent` as well. We
+      // need to remove them from the string so we can match it afterwards.
+      Array.from(node.querySelectorAll('textarea')).forEach(textarea => {
+        textToMatch = textToMatch.replace(textarea.value, '')
+      })
+
+      // The children of a select are also part of `textContent`, so we
+      // need also to remove their text.
+      Array.from(node.querySelectorAll('select')).forEach(select => {
+        textToMatch = textToMatch.replace(select.textContent, '')
+      })
+      return {node, textToMatch}
+    })
+    .filter(({textToMatch}) => textToMatch !== null)
+}
+
 function queryAllLabelsByText(
   container,
   text,
   {exact = true, trim, collapseWhitespace, normalizer} = {},
 ) {
   const matcher = exact ? matches : fuzzyMatches
   const matchNormalizer = makeNormalizer({collapseWhitespace, trim, normalizer})
-  return Array.from(container.querySelectorAll('label')).filter(label => {
-    let textToMatch = label.textContent
 
-    // The children of a textarea are part of `textContent` as well. We
-    // need to remove them from the string so we can match it afterwards.
-    Array.from(label.querySelectorAll('textarea')).forEach(textarea => {
-      textToMatch = textToMatch.replace(textarea.value, '')
-    })
+  const textToMatchByLabels = queryAllLabels(container)
+
+  const nodesByLabelMatchingText = textToMatchByLabels
+    .filter(({node, textToMatch}) =>
+      matcher(textToMatch, node, text, matchNormalizer),
+    )
+    .map(({node}) => node)
+  const labelsNotMatchingTextAndNotEmpty = textToMatchByLabels.filter(
+    ({node, textToMatch}) =>
+      Boolean(textToMatch) &&
+      nodesByLabelMatchingText.findIndex(nodeByLabelByText =>
+        nodeByLabelByText.isEqualNode(node),
+      ) === -1,
+  )
```

Comment:
```suggestion
  const labelsNotMatchingTextAndNotEmpty = textToMatchByLabels.filter(
    ({node, textToMatch}) =>
      textToMatch && nodesByLabelMatchingText.some(n => n.isEqualNode(node)),
  )
```

Reply:
I think the right change should be:
```javascript
const labelsNotMatchingTextAndNotEmpty = textToMatchByLabels.filter(
    ({node, textToMatch}) =>
      textToMatch && !nodesByLabelMatchingText.some(n => n.isEqualNode(node)),
  )
```
because I would like to retrieve all the labels in `textToMatchByLabels ` but not empty and not included in `nodesByLabelMatchingText`